vmdb/app/models/authenticator.rb

```diff
@@ -0,0 +1,263 @@
+module Authenticator
+  def self.for(config, username = nil)
+    subclass_for(config, username).new(config)
+  end
+
+  def self.subclass_for(config, username)
+    return Database if username == "admin"
+
+    case config[:mode]
+    when "database"      then Database
+    when "ldap", "ldaps" then Ldap
+    when "amazon"        then Amazon
+    when "httpd"         then Httpd
+    end
+  end
+
+  class Base
+    def self.authorize(config, *args)
+      new(config).authorize(*args)
+    end
+
+    def self.short_name
+      name.demodulize.underscore
+    end
+
+    attr_reader :config
+    def initialize(config)
+      @config = config
+    end
+
+    def password?
```

Comment:
Yup, that's fair. I was going for "is this a [local] password-based authentication mechanism".

`uses_password?`? `requires_password?`? When it's true, "supports" seems to be an understatement.